File: Abstraction/Ad-hoc/area.cpp
#include <iostream>

void print_area(int side) {
    std::cout << "Area of square is: " << side * side << '\n';
}

void print_area(int height, int width) {
    std::cout << "Area of rectangle is: " << height * width << '\n';
}

int main() {
    print_area(5);
    print_area(4, 6);
    return 0;
}

File: Abstraction/Ad-hoc/area.kt
fun printArea(side: Int) {
    println("Area of square is: ${side * side}")
}

fun printArea(height: Int, width: Int) {
    println("Area of rectangle is: ${height * width}")
}

fun main() {
    printArea(5)
    printArea(4, 6)
}
